src/modules/azure.rs

```diff
@@ -53,28 +74,44 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     Some(module)
 }
 
-fn get_azure_subscription_name(context: &Context) -> Option<SubscriptionName> {
+fn get_azure_profile_info(context: &Context) -> Option<Subscription> {
     let mut config_path = get_config_file_location(context)?;
     config_path.push("azureProfile.json");
 
-    let parsed_json = parse_json(&config_path)?;
-
-    let subscriptions = parsed_json.get("subscriptions")?.as_array()?;
-    let subscription_name = subscriptions.iter().find_map(|s| {
-        if s.get("isDefault")? == true {
-            Some(s.get("name")?.as_str()?.to_string())
+    if config_path.exists() {
+        let azure_profile: Option<AzureProfile> = load_azure_profile(&config_path);
+
+        if let Some(azure_profile) = azure_profile {
+            let subscription = azure_profile.subscriptions.iter().find_map(|s| {
+                if s.is_default {
+                    Some(s.clone())
+                } else {
+                    None
+                }
+            });
+            subscription
         } else {
             None
         }
-    });
-    if subscription_name.is_some() {
-        subscription_name
     } else {
-        log::info!("Could not find subscription name");
         None
     }
```

Comment:
Instead of checking for file existence, just try to open it:
```rust
    let azure_profile = load_azure_profile(&config_path)?;
    azure_profile
        .subscriptions
        .into_iter()
        .find(|s| s.is_default)
```